taurus/entity/object/tests/test_ingredient_spec.py

```diff
@@ -22,3 +29,56 @@ def test_ingredient_reassignment():
     assert potatoes.process == frying
     assert boiling.ingredients == []
     assert set(frying.ingredients) == {oil, potatoes}
+
+
+VALID_QUANTITIES = [
+    NominalReal(14.0, ''),
+    UniformReal(0.5, 0.6, 'm'),
+    NormalReal(-0.3, 0.6, "kg")
+]
+
+INVALID_QUANTITIES = [
+    NominalCategorical("blue"),
+    NominalInteger(5),
+    0.33,
+    "0.5"
```

Comment:
You may consider throwing an EmpiricalFormula in there, too

Reply:
Will do